crates/ruff_linter/src/rules/flake8_bandit/rules/django_extra.rs

```diff
@@ -0,0 +1,86 @@
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::{self as ast, Expr, ExprAttribute, ExprDict, ExprList};
+use ruff_python_semantic::Modules;
+use ruff_text_size::Ranged;
+
+use crate::checkers::ast::Checker;
+
+/// ## What it does
+/// Checks for uses of Django's `extra` function.
+///
+/// ## Why is this bad?
+/// Django's `extra` function can be used to execute arbitrary SQL queries,
+/// which can in turn lead to SQL injection vulnerabilities.
+///
+/// ## Example
+/// ```python
+/// from django.contrib.auth.models import User
+///
+/// User.objects.all().extra(select={'test': '%secure' % 'nos'})
+/// ```
+///
+/// ## References
+/// - [Django documentation: SQL injection protection](https://docs.djangoproject.com/en/dev/topics/security/#sql-injection-protection)
+/// - [Common Weakness Enumeration: CWE-89](https://cwe.mitre.org/data/definitions/89.html)
+#[violation]
+pub struct DjangoExtra;
+
+impl Violation for DjangoExtra {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Use of `extra` can lead to SQL injection vulnerabilities")
+    }
+}
+
+/// S610
+pub(crate) fn django_extra(checker: &mut Checker, call: &ast::ExprCall) {
+    if !checker.semantic().seen_module(Modules::DJANGO) {
+        return;
+    }
```

Comment:
Case in point, the ecosystem checks detects 18 errors while `bandit` detects 22. That's because there are 4 things that would have been flagged in https://github.com/zulip/zulip/blob/29ca4ba6620f16598e1171be7495356f56b84328/zerver/tests/test_message_move_topic.py, but are skipped because there is no import from `django` in this file.

Reply:
I think I'm okay with leaving this in for now.

Reply:
I could go either way though. `extra` isn't a common method name.

Reply:
Yeah, I think that the combination of `extra` and `select`/`where`/`tables` arguments should not be that common outside of Django. Worst case, for the few times it would get triggered, it's always possible for users to ignore the false positives, while skipping the check entirely if Ruff does not find a `django` import would leave the user with no clue that the check was not run because of that. So I think it might be best in the end to not check if Django was seen.

Reply:
I went ahead and removed the check in https://github.com/astral-sh/ruff/pull/10316/commits/4e881428c95628d03504b2d7ce5001256a2a26bf. Also updated the error message so that in case of false positives, the user at least knows that this is related to Django.